subsys/fb/cfb_shell.c

```diff
@@ -0,0 +1,411 @@
+/** @file
+ * @brief Monochrome Character Framebuffer shell module
+ *
+ * Provide some Character Framebuffer shell commands that can be useful for
+ * testing.
+ */
+
+/*
+ * Copyright (c) 2018 Diego Sueiro
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <shell/shell.h>
+#include <display/cfb.h>
+
+#define print(_sh, _ft, ...) \
+	shell_fprintf(_sh ? _sh : ctx_shell, SHELL_NORMAL, _ft "\r\n", \
+		      ##__VA_ARGS__)
+#define error(_sh, _ft, ...) \
+	shell_fprintf(_sh ? _sh : ctx_shell, SHELL_ERROR, _ft "\r\n", \
+		      ##__VA_ARGS__)
+
+#define HELP_NONE "[none]"
+#define HELP_INIT "call \"cfb init\" first"
+
+#if defined(CONFIG_SSD1673)
+#define DISPLAY_DRIVER				"SSD1673"
```

Comment:
This should come from Kconfig (e.g. CHARACTER_FRAMEBUFFER_SHELL_DISPLAY_NAME) now it limits the shell to only two drivers and further it assumes the names are hard coded to ```SSD1673``` and ```SSD1306```